api/server.js

```diff
@@ -27,14 +29,15 @@ server.get('/api/users', tokenCheck, async (req,res) => {
 server.post('/api/register', async (req,res) => {
     let user = req.body;
     if (!user.username || !user.password || !user.department) {
-        res.status(400).json({error:'please provide username/password'});
+        res.status(400).json({error:'please provide username/password/department'});
     } else {
         user.password = bcrypt.hashSync(user.password, 8);
         try {
-            const result = await db('users').insert(user);
-            if (result) {
-                const newUser = await db('users').where({'id':result[0]}).first();
-                res.status(201).json(newUser);
+            const [id] = await db('users').insert(user);
+            if (id) {
+                const newUser = await db('users').where({id}).first();
+                const token = generateToken(newUser);
```

Comment:
much more semantic naming of variables and array destructuring make this much more readable :+1: